src/main/java/uk/gov/hmcts/reform/feature/service/JwtParserService.java

```diff
@@ -0,0 +1,31 @@
+package uk.gov.hmcts.reform.feature.service;
+
+import io.jsonwebtoken.Claims;
+import io.jsonwebtoken.JwtParser;
+import uk.gov.hmcts.reform.feature.model.UserTokenDetails;
+
+import java.util.Arrays;
+import java.util.List;
+
+public class JwtParserService {
+
+    private final JwtParser parser;
+
+    public JwtParserService(JwtParser parser) {
+        this.parser = parser;
+    }
+
+    public UserTokenDetails parse(String jwtToken) {
+        try {
+            Claims claims = parser.parseClaimsJws(jwtToken).getBody();
+
+            String id = claims.get("id").toString();
+            List<String> roles = Arrays.asList(claims.get("data").toString().split(","));
```

Comment:
should this claim not be called `roles`?

Reply:
no it should not: https://github.com/hmcts/idam-app/blob/master/api/src/main/java/uk/gov/hmcts/idam/infrastructure/security/authentication/ClaimProperties.java#L4